spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1811,24 +1820,27 @@ trait OptimisticTransactionImpl extends TransactionalWrite
   protected def checkForConflictsAgainstVersion(
       currentTransactionInfo: CurrentTransactionInfo,
       otherCommitVersion: Long,
+      otherCommitFileStatus: FileStatus,
       commitIsolationLevel: IsolationLevel): CurrentTransactionInfo = {
 
     val conflictChecker = new ConflictChecker(
       spark,
       currentTransactionInfo,
       otherCommitVersion,
+      otherCommitFileStatus,
       commitIsolationLevel)
     conflictChecker.checkConflicts()
   }
 
   /** Returns the version that the first attempt will try to commit at. */
   protected def getFirstAttemptVersion: Long = readVersion + 1L
 
-  /** Returns the next attempt version given the last attempted version */
-  protected def getNextAttemptVersion(previousAttemptVersion: Long): Long = {
+  /** Returns the conflicting commit information */
+  protected def getConflictingVersions(previousAttemptVersion: Long): Seq[FileStatus] = {
+    assert(previousAttemptVersion == preCommitLogSegment.version + 1)
     val latestSnapshot = deltaLog.update()
     preCommitLogSegment = latestSnapshot.logSegment
-    latestSnapshot.version + 1
+    deltaLog.listFrom(previousAttemptVersion).filter(isDeltaFile).toSeq
```

Comment:
Doesn't this add extra listings to the conflict resolution path? 

The `deltaLog.update()` call just barely did the ~same listing two lines earlier; is there some way to reuse that work instead of repeating it?

Reply:
Good catch. Tried to reuse the same listing.